src/NServiceBus.RavenDB/Shutdown/ShutdownHook.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace NServiceBus.RavenDB.Shutdown
+{
+    /// <summary>
+    /// This class is to be ported to the Endpoint level in NServiceBus core.
+    /// This allows us to fix the timeout shutdown bug without also needing to patch core.
+    /// </summary>
+    class ShutdownHook : IWantToRunWhenBusStartsAndStops
```

Comment:
we should test this via Acceptance Tests running real endpoints, I have a very vague memory that in V4 it was registered as transient so it cannot be stateful and start and stop cannot share anything. If that's the case we can register in the container the `IContainShutdownDelegates` as well as singleton and depend on it